django_addanother/contrib/select2.py

```diff
@@ -21,13 +21,17 @@ def _gen_classes(globals_, locals_):
         'ModelSelect2TagWidget',
     ]
     cls_template = textwrap.dedent('''
+    class {new_widget_cls}({widget_cls}):
+        def optgroups(self, name, value, attrs=None):
+            ret = super().optgroups(name, value, attrs=attrs)
+            if not self.is_required and not self.allow_multiple_selected:
+                ret = ret[1:]
+            return ret
+
     class {new_cls_name}({wrapper_cls}):
         """:class:`~{widget_cls}` wrapped with :class:`~{wrapper_cls}`."""
         def __init__(self, *args, **kwargs):
-            super({new_cls_name}, self).__init__(
-                {widget_cls},
-                *args, **kwargs
-            )
+            super().__init__({new_widget_cls}, *args, **kwargs)
```

Comment:
We still support Python 2.7 so the old-style `super` needs to stay.

Reply:
Fixed!